src/services/activation/hooks/utils.js

```diff
@@ -0,0 +1,76 @@
+const { BadRequest, Forbidden } = require('@feathersjs/errors');
+const constants = require('../../../utils/constants');
+const { blockDisposableEmail } = require('../../../hooks');
+const { trimPassword } = require('../../account/hooks');
+const { checkUniqueAccount } = require('../../user/hooks/userService');
+const { hasPermission } = require('../../../hooks');
+const { externallyManaged } = require('../../helpers/utils');
+const logger = require('../../../logger');
+
+const nullOrEmpty = (string) => !string;
+
+const login = async (app, username, password, strategy = 'local') => app.service('authentication')
+	.create({
+		strategy,
+		username,
+		password,
+	})
+	.then((result) => result.accessToken)
+	.catch((error) => {
+		logger.error(error);
+		return null;
+	});
+
+const isValidLogin = (jwt) => !!jwt;
+
+const validPassword = async (hook) => {
+	if (!hook.data || !hook.data.password) throw new BadRequest('Missing information');
+	const { password } = hook.data;
+	const { username } = hook.params.account;
+
+	const jwt = await login(hook.app, username, password);
+	if (!isValidLogin(jwt)) {
+		throw new Forbidden('Not authorized');
+	}
+	return hook;
+};
+
+const blockThirdParty = async (hook) => {
+	const exm = await externallyManaged(hook.app, hook.params.account.userId);
+	if (exm) {
```

Comment:
```suggestion
	if (exm === true) {
```

do not use short terms